cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -92,7 +92,16 @@ type Etcd struct {
 	DataDir   string
 	ExtraArgs map[string]string
 	// Image specifies which container image to use for running etcd. If empty, automatically populated by kubeadm using the image repository and default etcd version
-	Image string
+	Image      string
+	SelfHosted SelfHostedEtcd
+}
+
+type SelfHostedEtcd struct {
+	CertificatesDir  string
```

Comment:
What's this for?

Reply:
It's where all TLS assets for etcd are stored. Because there are 8 files generated (2 CA files, 2 peers, 2 serving and 2 client certs), I thought it best to nest them in their own `etcd` directory inside the main CertDir. I thought having a new field would be a good idea because:

1. it provides a convenient way to retrieve this path. The alternative would be using `filepath.Join(cfg.CertificateDir, constants.EtcdTLSAssetsDir)` repeatedly across the codebase, or defining a function somewhere, like in the constants pkg.
2. it also allows configurability

Then again, if we feel this unnecessary, I can remove and replace with a func somewhere takes cfg and then returns the cert dir for etcd. WDYT?